packages/terra-clinical-data-grid/docs/README.md

```diff
@@ -0,0 +1,93 @@
+|**text**|string|optional|none|A string of text to render within the column header. This text will follow standard truncation rules as necessary. If a `component` value is specified, `text` will be ignored.|
+|**sortIndicator**|One of: **ascending**, **descending**|optional|none|A string indicating which sorting indicator should be rendered. If not provided, no sorting indicator will be rendered. If a `component` value is specified, `sortIndicator` will be ignored.
+|**component**|component|optional|none|A custom component to render within the column header. If provided, the `text` and `sortIndicator` values will be ignored.|
+
+### Sections
+
+Sections define the rendering style of the section header as well as the rows inside the section.
+
+Sections are provided to the DataGrid through the `sections` prop. The order in which the sections are provided will be the order in which they are rendered.
+
+|Name|Type|Is Required|Default Value|Description|
+|---|---|---|---|---|
+|**id**|string|required|none|An identifier for the section. This identifier should be unique across the set of sections provided to the DataGrid.|
+|**isCollapsible**|bool|optional|none|A boolean indicating whether or not the the section is collapsible. If true, the DataGrid's `onRequestSectionCollapse` function will be called upon selection of the section header.|
+|**isCollapsed**|bool|optional|none|A boolean indicating whether or not the section is collapsed. If true, the DataGrid will not render the contents of the section.|
+|**text**|string|optional|none|A text string to render within the section header.|
+|**startAccessory**|component|optional|none|A component to render within the section header's start accessory position. This will be rendered ahead of `text`.
+|**endAccessory**|component|optional|none|A component to render within the section header's end accessory position. This will be rendered behind `text`.
+|**component**|component|optional|none|A custom component to render within the section header. If provided, the `text`, `startAccessory`, and `endAccessory` values will be ignored.|
+|**rows**|array|optional|[]|An array of row objects to be rendered within the section.|
+
+> Note: if `text`, `startAccessory`, `endAccessory`, and `component` are not provided, and if `isCollapsible` is false, no section header will be rendered. However, the `rows` of the section will continue to be rendered.
+
+### Rows
+
+Rows define the cells rendered within the row as well as the row's selection properties.
+
+Rows are provided to sections to be rendered by the DataGrid. The order in which the rows are provided will be the order in which they are rendered.
+
+|Name|Type|Is Required|Default Value|Description|
```

Comment:
We could probably remove the props tables here and use the doc-template to generate information on each component.

Reply:
They're not components, they're just data objects. It'd be nice if the props table could take an arbitrary props object and spit out a table, but it'd be lacking descriptions at that point...